backend/app.js

```diff
@@ -33,19 +37,16 @@ app.use(function(req, res, next) {
 app.use(function(err, req, res, next) {
   // set locals, only providing error in development
   res.locals.message = err.message;
-  res.locals.error = req.app.get('env') === 'development' ? err : {};
+  res.locals.error = req.app.get("env") === "development" ? err : {};
 
   // render the error page
   res.status(err.status || 500);
-  res.render('error');
+  res.render("error");
 });
 
-// testing 
-const port = 3001;
-
-
+var port = process.env.port || 3000;
```

Comment:
Perhaps change the default port to 5000 to avoid clashes because React frontend uses 3000 as default.